src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -166,7 +170,14 @@ def convert_sp_to_sf_type(datatype: DataType) -> str:
     if isinstance(datatype, TimeType):
         return "TIME"
     if isinstance(datatype, TimestampType):
-        return "TIMESTAMP"
+        if datatype.tz == TimestampTimeZone.NTZ:
```

Comment:
This change doesn't seem handle infer_type https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/_internal/type_utils.py#L28j5, this is used at Snowpandas when the column is object type to derive the type of the column, right now it is mapped to Timestamp type, we probably need to further process this to set the tz information if it is timestamp type

Reply:
Great point! Let's chat offline and I'd like to understand more about the usage case. This can be tricky but also important.